cogip/tools/planner/planner.py

```diff
@@ -94,37 +100,105 @@ def add_robot(self, robot_id: int) -> None:
         """
         Add a new robot.
         """
-        if self._paths.get(robot_id) is None:
-            self._paths[robot_id] = available_paths.pop(0)
-        self._robot_pose[robot_id] = models.Pose()
-        self._pose_reached[robot_id] = True
+        if robot_id in self._robots:
+            self.del_robot(robot_id)
+        self._robots[robot_id] = (robot := Robot(robot_id, self._sio_ns))
+        robot.set_pose_start(self._game_context.get_start_pose(self._start_positions.get(robot_id, robot_id)))
+        self.update_start_pose_commands()
         self._obstacles[robot_id] = []
 
     def del_robot(self, robot_id: int) -> None:
         """
         Remove a robot.
         """
-        del self._pose_reached[robot_id]
-        del self._robot_pose[robot_id]
-        del self._obstacles[robot_id]
-        path = self._paths.pop(robot_id)
-        path.reset()
-        available_paths.append(path)
+        del self._robots[robot_id]
+        self.update_start_pose_commands()
+
+    def update_start_pose_commands(self):
+        """
+        Update menu commands to to choose start pose for each robot.
+        """
+        for robot_id, menu_cmd in self._start_pose_menu_entries.items():
+            menu.menu.entries.remove(menu_cmd)
+            if hasattr(self, cmd := f"cmd_choose_start_position_{robot_id}"):
+                delattr(self, cmd)
+        self._start_pose_menu_entries.clear()
+
+        for robot_id in sorted(self._robots.keys(), reverse=True):
+            func_name = f"choose_start_position_{robot_id}"
+            menu_entry = models.MenuEntry(
+                cmd=func_name,
+                desc=f"Choose Start Position {robot_id}"
+            )
+            menu.menu.entries.insert(0, menu_entry)
+            self._start_pose_menu_entries[robot_id] = menu_entry
+            setattr(self, "cmd_" + func_name, partial(self.cmd_choose_start_position, robot_id))
+            self._sio_ns.emit("register_menu", {"name": "planner", "menu": menu.menu.dict()})
+
+    def reset(self) -> None:
+        """
+        Reset planner, context, robot and actions.
+        """
+        self._game_context.reset()
+        robot_ids = list(self._robots.keys())
+        self._actions = actions.action_classes.get(self._game_context.strategy, actions.Actions)()
+
+        for robot_id in robot_ids:
+            self.del_robot(robot_id)
+        for robot_id in robot_ids:
+            self.add_robot(robot_id)
```

Comment:
You delete robots that you just re-add after.
not sure to understand, it deserves a comment I think.

Reply:
It is a simple solution to reset robots already registered.